### ISSUE TITLE
Automatic failover does not work even when running in interactive=false 

### ISSUE THREAD

**Issue by vipul-nextev** [signal18/replication-manager#98]
I am trying to setup automatic failover via maxscale.conf , but it does not work . Here is the input from maxscale.conf file 

[db_rw_router Monitor]
type=monitor
module=mysqlmon
servers=db1,db2,db3,db4
user=maxscale_user
passwd=xxxxx
detect_replication_lag=true
detect_stale_master=true
monitor_interval=500
script=/usr/bin/replication-manager  failover  --interactive=false
## events=master_down

When I turn off master . I see the output from list servers
# maxadmin -pxxxx list servers

Servers.
-------------------+-----------------+-------+-------------+--------------------
Server             | Address         | Port  | Connections | Status
-------------------+-----------------+-------+-------------+--------------------
db1                | 10.118.130.52   |  3306 |           0 | Down
db2                | 10.118.130.51   |  3306 |           0 | Slave, Running
db3                | 10.118.134.87   |  3306 |           0 | Down
db4                | 10.118.134.84   |  3306 |           0 | Slave, Running
-------------------+-----------------+-------+-------------+--------------------

Can you please help .. Maybe I am not doing something right

**Comment by vipul-nextev** [comment]
I even tried using /etc/replication-manager/config.toml file this is partial file 

autorejoin = true
readonly = true
wait-kill = 5000
post-failover-script = ""
pre-failover-script =""

check-type = "tcp"

failcount = 3
failover-limit = 0
failover-time-limit = 5
gtidcheck = true
maxdelay = 0
interactive = false
failtime=0
rplchecks=false

Thanks

**Comment by svaroqui** [comment]
Thanks for using replication-manager, can you provide us the replication manager logs ? You can enable more verbose log --verbose  --log-level=2

**Comment by svaroqui** [comment]
Can you tell also give us the --version

**Comment by tanji** [comment]
Please enable logs: --logfile=/tmp/mrm.log and attach to this issue

**Comment by vipul-nextev** [comment]
The log is enabled and I am running with --verbose and log-level = 2 

==> /var/log/maxscale/maxscale1.log <==
2016-10-11 17:49:26   error  : Monitor was unable to connect to server 10.118.130.51:3306 : "Can't connect to MySQL server on '10.118.130.51' (107)"
2016-10-11 17:49:26   notice : Server 10.118.130.51:3306 lost the master status.
2016-10-11 17:49:26   notice : Server changed state: db2[10.118.130.51:3306]: master_down. [Master, Running] -> [Down]
2016-10-11 17:49:26   notice : Executed monitor script '/usr/bin/replication-manager  failover  --interactive=false --verbose --log-level=2' on event 'master_down'.
2016-10-11 17:49:26   error  : No Master can be determined. Last known was 10.118.130.51:3306

==> /var/log/maxscale/replication-manager.log <==
2016/10/11 17:49:28 WARN : Master Failure detected! Retry 1/3

It seems like it is running in interactive mode even when I run it in interactive=false mode . 

Do I need to run replication manager service and mention the script in maxscale.conf ?

**Comment by svaroqui** [comment]
Well you can run replication-manager as a service   ./replication-manager monitor  --http-server 
in this case it will open http interface port 10001 where you can see what is happening or 

You can refer to this blog post 
https://mariadb.com/blog/mariadb-high-availability-replication-manager

In such describe use case you should  comment any maxscale scripts , you have been previously set like :  
script=/usr/bin/replication-manager failover --interactive=false

Tell us if it helps

**Comment by svaroqui** [comment]
the default is   interactive =false : failover will be manual , and an email should be sent to you to let you know to connect to http server to finish failover. 
You can change the mode from interface if you wan't . 
Last release should also drive maxscale to put the old master in maintenance 

Guillaume Tanji can tell you more how to use that feature

**Comment by vipul-nextev** [comment]
when I set replication-manager script in maxscale.conf whether running in interactive=true or interactive=false failover does not happen automatically. I tried with maxscale 1.4 and maxscale 2.0 

==> /var/log/maxscale/maxscale1.log <==
2016-10-11 22:22:23   error  : Monitor was unable to connect to server 10.118.130.51:3306 : "Can't connect to MySQL server on '10.118.130.51' (107)"
2016-10-11 22:22:23   notice : Server 10.118.130.51:3306 lost the master status.
2016-10-11 22:22:23   notice : Server changed state: db2[10.118.130.51:3306]: master_down. [Master, Running] -> [Down]
2016-10-11 22:22:23   notice : Executed monitor script '/usr/bin/replication-manager  failover  --interactive=true --verbose --log-level=2' on event 'master_down'.
2016-10-11 22:22:23   error  : No Master can be determined. Last known was 10.118.130.51:3306

2016-10-11 22:23:14   notice : Loaded module MaxAdminAuth: V2.0.0 from /usr/lib64/maxscale/libMaxAdminAuth.so
^C
[root@ip-10-118-130-62 ~]# maxadmin -pmariadb list servers
Servers.
-------------------+-----------------+-------+-------------+--------------------
Server             | Address         | Port  | Connections | Status
-------------------+-----------------+-------+-------------+--------------------
db1                | 10.118.130.52   |  3306 |           0 | Slave, Running
db2                | 10.118.130.51   |  3306 |           0 | Down
db4                | 10.118.134.84   |  3306 |           0 | Slave, Running
-------------------+-----------------+-------+-------------+--------------------

Can you please point me to a script which has automatic failover working , so I can copy and try it .

Thanks

**Comment by tanji** [comment]
@vipul-nextev please give us the output of `replication-manager version`. Thanks in advance

Your script should work as is. Failover doesn't care whether interactive is set or not.

**Comment by vipul-nextev** [comment]
replication-manager version
MariaDB Replication Manager version 0.7
Version:
Build Time:

**Comment by vipul-nextev** [comment]
I have compiled replication-manager binary couple of weeks ago but something has changed since then . I recompiled my binary today using the documentation from 
https://mariadb.com/blog/mariadb-high-availability-replication-manager.

One thing I notice is when I start the my older replicaiton-manager the log says 
2016/10/12 19:10:04 INFO : replication-manager version 0.7 started in daemon mode

And when failure is detected it says 
2016/10/12 19:12:56 WARN : Master Failure detected! Retry 1/5 
an it seems to get stuck .

In the newer binary , it says 
replication-manager version 0.7 started in  automatic mode.

and when it detects the failure , it does not get stuck at retry level ... FYI.

So bottom line , with current version it seems to work fine . 

Thanks for your help
